quickstarts/microsoft.dbforpostgresql/flexible-postgresql-with-vnet-dataencryption/main.bicep

```diff
@@ -0,0 +1,241 @@
+
+@description('Subnet Name')
+param subnetName string = 'azure_postgresql_subnet'
+
+@description('Virtual Network Address Prefix')
+param vnetAddressPrefix string = '10.0.0.0/24'
+
+@description('Subnet Address Prefix')
+param postgresqlSubnetPrefix string = '10.0.0.0/28'
+
+@description('Composing the subnetId')
+var postgresqlSubnetId =  '${vnetLink.properties.virtualNetwork.id}/subnets/${subnetName}'
+
+@description('Name of the managed identity resource')
+param userAssignedIdentityName string = 'azure_postgresql_userassignedidentity_1'
+
+@description('Specifies the name of the key vault.')
+param keyVaultName string
+
+@description('Specifies the name of the key in key vault.')
+param keyName string
+
+@description('Specifies the Azure Active Directory tenant ID that should be used for authenticating requests to the key vault. Get it by using Get-AzSubscription cmdlet.')
+param tenantId string = subscription().tenantId
+
+@description('Specifies the permissions to keys in the vault. Valid values are: all, encrypt, decrypt, wrapKey, unwrapKey, sign, verify, get, list, create, update, import, delete, backup, restore, recover, and purge.')
+param keysPermissions array = [
+  'get'
+  'list'
+  'wrapKey'
+  'unwrapKey'
+]
+
+@description('Specifies the permissions to secrets in the vault. Valid values are: all, get, list, set, delete, backup, restore, recover, and purge.')
+param secretsPermissions array = [
+  'list'
+]
+
+@description('String array containing any of: decrypt, encrypt, import, release, sign, unwrapKey, verify, wrapKey')
+param keyOps array = [
+  'decrypt'
+  'encrypt'
+  'sign'
+  'unwrapKey'
+  'verify'
+  'wrapKey'
+]
+
+@description('High Availability Mode')
+@allowed([
+  'Disabled'
+  'ZoneRedundant'
+  'SameZone'
+])
+param haMode string = 'Disabled'
+
+resource managedidentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: userAssignedIdentityName
+  location: location
+}
+
+resource kv 'Microsoft.KeyVault/vaults@2022-07-01' = {
+  name: keyVaultName
+  location: location
+  properties: {
+    tenantId: tenantId
+    accessPolicies: [
+      {
+        objectId: managedidentity.properties.principalId
+        tenantId: tenantId
+        permissions: {
+          keys: keysPermissions
+          secrets: secretsPermissions
+        }
+      }
+    ]
+    enablePurgeProtection: true
+    enableSoftDelete: true
+    sku: {
+      name: 'standard'
+      family: 'A'
+    }
+  }
+}
+
+resource kvKey 'Microsoft.KeyVault/vaults/keys@2022-07-01' = {
+  name: keyName
+  parent: kv
+  properties: {
+    attributes: {
+      enabled: true
+    }
+    keyOps: keyOps
+    keySize: 2048
+    kty: 'RSA'
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2022-07-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+  }
+
+resource subnet 'subnets@2022-07-01' = {
```

Comment:
Deploying subnets as a child resource will cause issues during re-deployment. Instead, create subnets via the `subnets` property in the vnet resource.